docs/pages/upgrading/agent-managed-updates.mdx

```diff
@@ -0,0 +1,512 @@
+---
+
+<Notice type="warning">
+This document describes the Managed Updates v2 updater, which is currently in beta.
+
+For Managed Updates v1 instructions, see
+[Managed Updates for Agents (v1)](./agent-managed-updates-v1.mdx).
+</Notice>
+
+Managed updates v2 use a binary called `teleport-update` which is distributed in
+all Teleport packages, alongside Teleport. The updates are configured by the
+`autoupdate_version` and `autoupdate_config` resources.
+
+This document covers how to use `teleport-update` and the `autoupdate_*`
+resources to manage your agent updates from Teleport.
+
+`teleport-update` supports:
+- Every Teleport edition (Enterprise, Community, OSS)
+- both cloud and self-hosted Teleport deployments
+- Regular and FIPS variants of Teleport
+- amd64 and arm64 CPU architectures
+- systemd-based OSes, regardless of the package manager used
+
+<Admonition type="note" title="Compatibility between Managed Updates v1 and v2">
+Managed updates v2 are backwards compatible with the
+`cluster_maintenance_config` resource. The Managed Updates v1 `teleport-upgrade` script
+is forwards compatible with the `autoupdate_config` and `autoupdate_version` resources.
+Agents connected to the same cluster should all update to the same version.
+
+If the `autoupdate_version` resource is configured, it takes precedence over
+`cluster_maintenance_config`. This allows for a safe, non-breaking, incremental
+migration between Managed Updates v1 and v2.
+
+Users of cloud-hosted Teleport Enterprise will be migrated to Managed Updates v2
+in the first half of 2025 and should plan to migrate their agents to `teleport-update`.
+</Admonition>
+
+## How it works
+
+When Managed Updates are enabled, a Teleport updater is installed alongside
+each new Teleport Agent. The updater communicates with the Teleport Proxy Service to
+determine when an update is available and if it should perform the update now.
+
+Each agent belongs to an update group. The update schedule specifies when each
+group is updated. The schedule is stored in the `autoupdate_config` resource and
+can be edited via `tctl`.
+
+For Linux server-based installations, `teleport-update` command configures
+Managed Updates locally on the server.
+
+For Kubernetes-based installations, the `teleport-kube-agent` Helm chart
+deploys a controller that automatically updates the main Teleport container.
+
+Existing agents must be manually enrolled into Managed Updates.
+
+## Prerequisites
+
+- A Teleport cluster. If you do not have one, [sign
+  up](https://goteleport.com/signup) for a free trial or consult the
+  [Teleport Installation](../installation.mdx) page.
+- Familiarity with the [Upgrading Compatibility Overview](./overview.mdx) guide,
+  which describes the sequence in which to upgrade components in your cluster.
+- Teleport Agents that are not yet enrolled in Managed Updates.
+- (!docs/pages/includes/tctl-tsh-prerequisite.mdx!)
+- (!docs/pages/includes/tctl.mdx!)
+
```

Comment:
Since the H2 sections for this guide are instructions for different scenarios, rather than a single sequence of high-level steps, I would add some text here linking the reader to the different sections depending on their needs. Otherwise, a user has to work their way through the whole guide in sequence, which doesn't seem like the intention.